src/Raven.Client/Documents/Linq/LinqPathProvider.cs

```diff
@@ -202,17 +202,8 @@ public MemberExpression GetMemberExpression(Expression expression)
             if (lambdaExpression != null)
                 return GetMemberExpression(lambdaExpression.Body);
 
-            var memberExpression = expression as MemberExpression;
 
-            if (memberExpression == null)
```

Comment:
I'm pretty sure that this is wrong, in the sense that we still want this error _during where_, but not on the select.